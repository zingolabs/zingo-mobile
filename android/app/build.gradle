apply plugin: "com.android.application"
apply plugin: "com.facebook.react"
apply plugin: 'kotlin-android'

/**
 * This is the configuration block to customize your React Native Android app.
 * By default you don't need to apply any configuration, just uncomment the lines you need.
 */

react {
    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    //   The folder where the react-native Codegen package is. Default is ../node_modules/@react-native/codegen
    // codegenDir = file("../node_modules/@react-native/codegen")
    //   The cli.js file which is the React Native CLI entrypoint. Default is ../node_modules/react-native/cli.js
    // cliFile = file("../node_modules/react-native/cli.js")
    /* Variants */
    //   The list of variants to that are debuggable. For those we're going to
    //   skip the bundling of the JS bundle and the assets. By default is just 'debug'.
    //   If you add flavors like lite, prod, etc. you'll have to list your debuggableVariants.
    // debuggableVariants = ["liteDebug", "prodDebug"]
    /* Bundling */
    //   A list containing the node command and its flags. Default is just 'node'.
    // nodeExecutableAndArgs = ["node"]
    //
    //   The command to run when bundling. By default is 'bundle'
    // bundleCommand = "ram-bundle"
    //
    //   The path to the CLI configuration file. Default is empty.
    // bundleConfig = file(../rn-cli.config.js)
    //
    //   The name of the generated asset file containing your JS bundle
    // bundleAssetName = "MyApplication.android.bundle"
    //
    //   The entry file for bundle generation. Default is 'index.android.js' or 'index.js'
    // entryFile = file("../js/MyApplication.android.js")
    //
    //   A list of extra flags to pass to the 'bundle' commands.
    //   See https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle
    // extraPackagerArgs = []
    /* Hermes Commands */
    //   The hermes compiler command to run. By default it is 'hermesc'
    // hermesCommand = "$rootDir/my-custom-hermesc/bin/hermesc"
    //
    //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
    // hermesFlags = ["-O", "-output-source-map"]
}

/**
 * Set this to true to Run Proguard on Release builds to minify the Java bytecode.
 */
def enableProguardInReleaseBuilds = false

/**
 * The preferred build flavor of JavaScriptCore (JSC)
 *
 * For example, to use the international variant, you can use:
 * `def jscFlavor = 'org.webkit:android-jsc-intl:+'`
 *
 * The international variant includes ICU i18n library and necessary data
 * allowing to use e.g. `Date.toLocaleString` and `String.localeCompare` that
 * give correct results when using with locales other than en-US.  Note that
 * this variant is about 6MiB larger per architecture than default.
 */
def jscFlavor = 'org.webkit:android-jsc:+'

/**
 * Whether to enable building a separate APK for each ABI.
 *
 * Defaults to false but can be set to true with the project properties flag
 * e.g. ./gradlew assembleDebug -PsplitApk=true
 *
 * Additional option to include a universal APK
 * e.g. ./gradlew assembleRelease -PsplitApk=true -PincludeUniversalApk=true
 */
ext.splitApk = project.properties['splitApk'] ?: false
ext.includeUniversalApk = project.properties['includeUniversalApk'] ?: false

android {
    namespace 'org.ZingoLabs.Zingo'
    ndkVersion rootProject.ext.ndkVersion

    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility = "18.0.2.1"
        targetCompatibility = "18.0.2.1"
    }

    kotlinOptions {
        jvmTarget = '18'
    }

    defaultConfig {
        applicationId 'org.ZingoLabs.Zingo' // Real
        //applicationId 'com.ZingoMobile' // @Test
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 126 // Real
        //versionCode 117 // @Test
        versionName "zingo-1.2.5" // Real
        missingDimensionStrategy 'react-native-camera', 'general'
        testBuildType System.getProperty('testBuildType', 'debug')
        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
    }

    splits {
        abi {
            enable splitApk.toBoolean()
            reset()
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
            universalApk includeUniversalApk.toBoolean()
        }
    }

    signingConfigs {
        debug {
            storeFile file('debug.keystore')
            storePassword 'android'
            keyAlias 'androiddebugkey'
            keyPassword 'android'
        }
    }

    packagingOptions {
        jniLibs {
            pickFirsts += ['lib/armeabi-v7a/libc++_shared.so', 'lib/arm64-v8a/libc++_shared.so', 'lib/x86/libc++_shared.so', 'lib/x86_64/libc++_shared.so']
        }
    }

    buildTypes {
        debug {
            signingConfig signingConfigs.debug
        }
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://reactnative.dev/docs/signed-apk-android.
            signingConfig signingConfigs.debug
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            proguardFile "${rootProject.projectDir}/../node_modules/detox/android/detox/proguard-rules-app.pro"
        }
    }

    buildFeatures {
        viewBinding = true
    }

    testOptions {
        // Managed devices for gradle integration testing
        managedDevices {
            devices {
                pixel2api29_x86 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 29
                    systemImageSource = "aosp"
                    require64Bit = false
                }
                pixel2api30_x86_64 (com.android.build.api.dsl.ManagedVirtualDevice) {
                    device = "Pixel 2"
                    apiLevel = 30
                    systemImageSource = "aosp"
                    require64Bit = true
                }
            }
            groups {
                x86_Archs {
                    targetDevices.add(devices.pixel2api29_x86)
                    targetDevices.add(devices.pixel2api30_x86_64)
                }
            }
        }
    }
    
    // // applicationVariants are e.g. debug, release
    // applicationVariants.all { variant ->
    //     variant.outputs.each { output ->
    //         // For each separate APK per architecture, set a unique version code as described here:
    //         // https://developer.android.com/studio/build/configure-apk-splits.html
    //         // Example: versionCode 1 will generate 1001 for armeabi-v7a, 1002 for x86, etc.
    //         def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
    //         def abi = output.getFilter(OutputFile.ABI)
    //         if (abi != null) {  // null for the universal-debug, universal-release variants
    //             output.versionCodeOverride =
    //                     defaultConfig.versionCode * 1000 + versionCodes.get(abi)
    //         }
    //     }
    // }
}


// Map for the version code that gives each ABI a value.
ext.abiCodes = ['armeabi-v7a':1, 'x86':2, 'arm64-v8a': 3, 'x86_64':4]

import com.android.build.OutputFile

// For each APK output variant, override versionCode with a combination of
// ext.abiCodes * 10000 + variant.versionCode. 
// variant.versionCode is equal to defaultConfig.versionCode.
// If you configure product flavors that define their own versionCode, 
// variant.versionCode uses that value instead.
android.applicationVariants.all { variant ->
  variant.outputs.each { output ->
    def baseAbiVersionCode = project.ext.abiCodes.get(output.getFilter(OutputFile.ABI))

    // Because abiCodes.get() returns null for ABIs that are not mapped by ext.abiCodes,
    // the following code doesn't override the version code for universal APKs.
    // However, because you want universal APKs to have the lowest version code,
    // this outcome is desirable.
    if (baseAbiVersionCode != null) {

      // Assigns the new version code to versionCodeOverride, which changes the
      // version code for only the output APK, not for the variant itself. Skipping
      // this step causes Gradle to use the value of variant.versionCode for the APK.
      output.versionCodeOverride = baseAbiVersionCode * 10000 + variant.versionCode
    }
  }
}
            
dependencies {
    // The version of react-native is set by the React Native Gradle Plugin
    implementation("com.facebook.react:react-android")
    androidTestImplementation('com.wix:detox:+')
    implementation 'androidx.appcompat:appcompat:1.1.0'
    implementation fileTree(dir: "libs", include: ["*.jar"])

    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
    implementation 'com.facebook.soloader:soloader:0.10.5+'

    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }

    // Detox tests getAttributes() needs this
    debugImplementation('com.google.android.material:material:1.2.1')

    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
        exclude group:'com.squareup.okhttp3', module:'okhttp'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

    if (hermesEnabled) {
        implementation("com.facebook.react:hermes-engine:+") {
            exclude group:'com.facebook.fbjni'
        }
    } else {
        implementation jscFlavor
    }
	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"

    def work_version = "2.7.1"

    // (Java only)
    implementation "androidx.work:work-runtime:$work_version"

    // Kotlin + coroutines
    implementation "androidx.work:work-runtime-ktx:$work_version"

    // optional - RxJava2 support
    implementation "androidx.work:work-rxjava2:$work_version"

    // optional - GCMNetworkManager support
    implementation "androidx.work:work-gcm:$work_version"

    // optional - Test helpers
    androidTestImplementation "androidx.work:work-testing:$work_version"

    // optional - Multiprocess support
    implementation "androidx.work:work-multiprocess:$work_version"
    
    // google truth testing framework
    androidTestImplementation "com.google.truth:truth:1.1.3"
    
    // JSON parsing
    implementation "com.fasterxml.jackson.module:jackson-module-kotlin:2.14.+"
    
    // JUnit test runners
    androidTestImplementation 'junit:junit:4.13'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.implementation
    into 'libs'
}

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
repositories {
    mavenCentral()
    google()
}
