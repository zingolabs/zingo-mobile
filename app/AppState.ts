// new class with all the information about the sync process
export class SyncStatusReport {
  // sync ID
  syncID: number;

  // total of batches to process (usually 1000 blocks per batch)
  totalBatches: number;

  // batch that is processing now
  currentBatch: number;

  // total blocks per batch
  blocksPerBatch: number;

  // last block of the wallet
  lastBlockWallet: number;

  // block that is processing
  currentBlock: number;

  inProgress: boolean;

  lastError: string;

  secondsPerBatch: number;

  percent: number;

  message: string;

  process_end_block: number;

  lastBlockServer: number;

  constructor() {
    this.syncID = 0;
    this.totalBatches = 0;
    this.currentBatch = 0;
    this.blocksPerBatch = 1000; // this is set in zingolib
    this.lastBlockWallet = 0;
    this.currentBlock = 0;
    this.inProgress = false;
    this.lastError = '';
    this.secondsPerBatch = 0;
    this.percent = 0;
    this.message = '';
    this.process_end_block = 0;
    this.lastBlockServer = 0;
  }
}

export class TotalBalance {
  // Total t address, confirmed and spendable
  transparentBal: number;

  // Total private, confirmed + unconfirmed
  privateBal: number;

  // Total orchard, confirmed + unconfirmed
  orchardBal: number;

  // Total private, confirmed funds that are spendable
  spendablePrivate: number;

  // Total orchard, confirmed funds that are spendable
  spendableOrchard: number;

  // Total unconfirmed + spendable
  total: number;

  constructor() {
    this.transparentBal = 0;
    this.privateBal = 0;
    this.spendablePrivate = 0;
    this.orchardBal = 0;
    this.spendableOrchard = 0;
    this.total = 0;
  }
}

export class Address {
  uaAddress: string;
  address: string;
  addressKind: string;
  containsPending: boolean;

  constructor(uaAddress: string, address: string, addressKind: string) {
    this.uaAddress = uaAddress;
    this.address = address;
    this.addressKind = addressKind;
    this.containsPending = false;
  }
}

export class AddressBookEntry {
  label: string;

  address: string;

  constructor(label: string, address: string) {
    this.label = label;
    this.address = address;
  }
}

export class SettingsFileEntry {
  server?: string;

  constructor(server: string) {
    this.server = server;
  }
}

export interface TxDetailType {
  address: string;
  amount: number;
  memo: string | null;
}

// List of transactions.
export interface Transaction {
  type: string;
  address: string;
  amount: number;
  position: string;
  confirmations: number;
  txid: string;
  time: number;
  zec_price: number | null;
  detailedTxns: TxDetailType[];
}

// This is the type that the RPC interface expects for sending
export interface SendJsonToType {
  address: string;
  amount: number;
  memo?: string;
}

export class ToAddr {
  id: number;
  to: string;
  amount: string;
  amountUSD: string;
  memo: string;

  constructor(id: number) {
    this.id = id;

    this.to = '';
    this.amount = '';
    this.amountUSD = '';
    this.memo = '';
  }
}

export class SendPageState {
  toaddrs: ToAddr[];

  constructor() {
    this.toaddrs = [];
  }
}

export class ReceivePageState {
  // A newly created address to show by default
  newAddress: string;

  // The key used for the receive page component.
  // Increment to force re-render
  rerenderKey: number;

  constructor() {
    this.newAddress = '';
    this.rerenderKey = 0;
  }
}

export interface InfoType {
  chain_name: string;
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  zecPrice: number | null;
  defaultFee: number;
}

export class SendProgress {
  sendInProgress: boolean;
  progress: number;
  total: number;
  etaSeconds: number | string;

  constructor() {
    this.sendInProgress = false;
    this.progress = 0;
    this.total = 0;
    this.etaSeconds = 0;
  }
}

export interface ServerSelectState {
  modalIsOpen: boolean;
}

export class PasswordState {
  showPassword: boolean;

  confirmNeeded: boolean;

  passwordCallback: ((password: string) => void) | null;

  closeCallback: (() => void) | null;

  helpText: string | null;

  constructor() {
    this.showPassword = false;
    this.confirmNeeded = false;
    this.passwordCallback = null;
    this.closeCallback = null;
    this.helpText = null;
  }
}

export class ErrorModalData {
  title: string;
  body: string;

  modalIsOpen: boolean;

  constructor() {
    this.modalIsOpen = false;
    this.title = '';
    this.body = '';
  }
}

export interface ConnectedCompanionApp {
  name: string;
  lastSeen: number;
}

export interface WalletSeed {
  seed: string;
  birthday: number;
}

export interface SyncStatus {
  inProgress: boolean;
  progress: number;
  blocks: string;
}

export class WalletSettings {
  download_memos: string;
  transaction_filter_threshold: string;
  server?: string;

  constructor() {
    this.download_memos = 'wallet';
    this.transaction_filter_threshold = '500';
  }
}

export default interface AppState {
  // Info about the current sync process
  syncStatusReport: SyncStatusReport;

  // The total confirmed and unconfirmed balance in this wallet
  totalBalance: TotalBalance;

  // A map type that contains address -> privatekey mapping, for display on the receive page
  // This mapping is ephemeral, and will disappear when the user navigates away.
  addressPrivateKeys: Map<string, string>;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: Address[];

  // List of Address / Label pairs
  addressBook: AddressBookEntry[];

  // List of all T and Z and O transactions
  transactions: Transaction[] | null;

  // The state of the send page, as the user constructs a transaction
  sendPageState: SendPageState;

  // Any state for the receive page
  receivePageState: ReceivePageState;

  // getinfo and getblockchaininfo result
  info: InfoType | null;

  // Is the app rescanning?
  rescanning: boolean;

  // Callbacks for the password dialog box
  //passwordState: PasswordState;

  wallet_settings: WalletSettings;

  syncingStatus: SyncStatus | null;

  // Data for any error or info modal
  errorModalData: ErrorModalData;

  // Build progress from Tx
  txBuildProgress: SendProgress;

  walletSeed: WalletSeed | null;

  isMenuDrawerOpen: boolean;

  selectedMenuDrawerItem: string;

  aboutModalVisible: boolean;

  computingModalVisible: boolean;

  settingsModalVisible: boolean;

  infoModalVisible: boolean;

  rescanModalVisible: boolean;

  seedViewModalVisible: boolean;
  seedChangeModalVisible: boolean;
  seedBackupModalVisible: boolean;
  seedServerModalVisible: boolean;

  syncReportModalVisible: boolean;

  newServer: string | null;

  uaAddress: string | null;

  // eslint-disable-next-line semi
}
