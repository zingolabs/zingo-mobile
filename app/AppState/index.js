import AddressClass from './classes/AddressClass';
import AddressBookClass from './classes/AddressBookClass';
import ErrorModalDataClass from './classes/ErrorModalDataClass';
import ReceivePageStateClass from './classes/ReceivePageStateClass';
import SendPageStateClass from './classes/SendPageStateClass';
import SendProgressClass from './classes/SendProgressClass';
import SettingsFileClass from './classes/SettingsFileClass';
import SyncingStatusClass from './classes/SyncingStatusClass';
import ToAddrClass from './classes/ToAddrClass';
import TotalBalanceClass from './classes/TotalBalanceClass';
import WalletSettingsClass from './classes/WalletSettingsClass';

import InfoType from './types/InfoType';
import SendJsonToTypeType from './types/SendJsonToTypeType';
import TransactionType from './types/TransactionType';
import TxDetailType from './types/TxDetailType';
import WalletType from './types/WalletType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundType from './types/BackgroundType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import ZecPriceType from './types/ZecPriceType';
import FetchErrorType from './types/FetchErrorType';

import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';

export {
  AppStateLoaded,
  AppStateLoading,
  AddressClass,
  AddressBookClass,
  ErrorModalDataClass,
  ReceivePageStateClass,
  SendPageStateClass,
  SendProgressClass,
  SettingsFileClass,
  SyncingStatusClass,
  ToAddrClass,
  TotalBalanceClass,
  WalletSettingsClass,
  TransactionType,
  TxDetailType,
  WalletType,
  SendJsonToTypeType,
  InfoType,
  ZecPriceType,
  BackgroundType,
  BackgroundErrorType,
  TranslateType,
  NetInfoType,
  ServerType,
  SnackbarType,
  FetchErrorType,
};
