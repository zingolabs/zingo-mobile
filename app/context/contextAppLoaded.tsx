import React, { ReactNode } from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import {
  AppStateLoaded,
  SyncingStatusClass,
  TotalBalanceClass,
  ReceivePageStateClass,
  ErrorModalDataClass,
  SendProgressClass,
  WalletSettingsClass,
  TransactionType,
  NoteType,
  InfoType,
  WalletType,
  AddressClass,
  AddressBookClass,
  ZecPriceType,
  BackgroundType,
  SendPageStateClass,
  ToAddrClass,
  NetInfoType,
  BackgroundErrorType,
  ServerType,
  SnackbarType,
  FetchErrorType,
} from '../AppState';

export const defaultAppStateLoaded: AppStateLoaded = {
  navigation: {} as StackScreenProps<any>['navigation'],
  route: {} as StackScreenProps<any>['route'],
  appState: '',
  netInfo: {} as NetInfoType,

  syncingStatus: new SyncingStatusClass(),
  totalBalance: new TotalBalanceClass(),
  addressPrivateKeys: new Map(),
  addresses: [] as AddressClass[],
  addressBook: [] as AddressBookClass[],
  transactions: [] as TransactionType[],
  notes: [] as NoteType[],
  sendPageState: new SendPageStateClass(new ToAddrClass(0)),
  receivePageState: new ReceivePageStateClass(''),
  info: {} as InfoType,
  walletSettings: new WalletSettingsClass(),
  errorModalData: new ErrorModalDataClass('', ''),
  sendProgress: new SendProgressClass(0, 0, 0),
  wallet: {} as WalletType,
  isMenuDrawerOpen: false,
  selectedMenuDrawerItem: '',
  aboutModalVisible: false,
  computingModalVisible: false,
  settingsModalVisible: false,
  infoModalVisible: false,
  rescanModalVisible: false,
  seedViewModalVisible: false,
  seedChangeModalVisible: false,
  seedBackupModalVisible: false,
  seedServerModalVisible: false,
  ufvkViewModalVisible: false,
  ufvkChangeModalVisible: false,
  ufvkBackupModalVisible: false,
  ufvkServerModalVisible: false,
  syncReportModalVisible: false,
  poolsModalVisible: false,
  insightModalVisible: false,
  newServer: {} as ServerType,
  uaAddress: '',

  server: {} as ServerType,
  currency: '',
  language: 'en',

  zecPrice: {
    zecPrice: 0,
    date: 0,
  } as ZecPriceType,
  sendAll: false,
  background: {
    batches: 0,
    date: 0,
  } as BackgroundType,

  translate: () => '',
  backgroundError: {} as BackgroundErrorType,
  setBackgroundError: () => {},
  privacy: false,
  readOnly: false,
  poolsToShieldSelectSapling: true,
  poolsToShieldSelectTransparent: true,
  mode: 'basic',
  snackbars: [] as SnackbarType[],
  addLastSnackbar: () => {},
  restartApp: () => {},
  someUnconfirmed: false,
  fetchError: {} as FetchErrorType,
  setFetchError: () => {},
};

export const ContextAppLoaded = React.createContext(defaultAppStateLoaded);

type ContextProviderProps = {
  children: ReactNode;
  value: AppStateLoaded;
};

export const ContextAppLoadedProvider = ({ children, value }: ContextProviderProps) => {
  return <ContextAppLoaded.Provider value={value}>{children}</ContextAppLoaded.Provider>;
};
