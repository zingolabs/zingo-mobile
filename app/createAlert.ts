import AsyncStorage from '@react-native-async-storage/async-storage';
import { Alert } from 'react-native';
import SnackbarType from './AppState/types/SnackbarType';
import { GlobalConst, TranslateType } from './AppState';

export const createAlert = async (
  setBackgroundError: (title: string, error: string) => void,
  addLastSnackbar: (snackbar: SnackbarType) => void,
  title: string,
  error: string,
  toast: boolean,
  translate: (key: string) => TranslateType,
  sendEmail?: (translate: (key: string) => TranslateType, z: string, s?: string, b?: string) => void,
  zingolibVersion?: string,
) => {
  const background = await AsyncStorage.getItem(GlobalConst.background);
  if (background === GlobalConst.yes) {
    setBackgroundError(title, error);
  } else {
    if (toast) {
      setTimeout(() => {
        addLastSnackbar({ message: error });
      }, 1000);
    } else {
      if (sendEmail) {
        // with email button
        Alert.alert(
          title,
          error,
          [
            {
              text: translate('support') as string,
              onPress: async () => sendEmail(translate, zingolibVersion ? zingolibVersion : '', title, error),
            },
            { text: translate('cancel') as string, style: 'cancel' },
          ],
          { cancelable: false, userInterfaceStyle: 'light' },
        );
      } else {
        // no email button
        Alert.alert(title, error);
      }
    }
  }
};
