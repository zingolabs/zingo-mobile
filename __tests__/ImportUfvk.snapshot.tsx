/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import { ImportUfvk } from '../components/Ufvk';
import { ContextAppLoadedProvider, defaultAppStateLoaded } from '../app/context';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.',
      groupingSeparator: ',',
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('react-native-reanimated', () => {
  return class Reanimated {
    public static Value() {
      return jest.fn(() => {});
    }
    public static View() {
      return '';
    }
  };
});
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
    getLatestBlock: jest.fn(() => '{}'),
  };

  return RN;
});

// test suite
describe('Component ImportUfvk - test', () => {
  //snapshot test
  test('ImportUfvk - snapshot', () => {
    const state = defaultAppStateLoaded;
    state.translate = () => 'text translated';
    state.info.currencyName = 'ZEC';
    state.totalBalance.total = 1.12345678;
    const onCancel = jest.fn();
    const onOK = jest.fn();
    const importUfvk = render(
      <ContextAppLoadedProvider value={state}>
        <ImportUfvk onClickCancel={onCancel} onClickOK={onOK} />
      </ContextAppLoadedProvider>,
    );
    expect(importUfvk.toJSON()).toMatchSnapshot();
  });
});
