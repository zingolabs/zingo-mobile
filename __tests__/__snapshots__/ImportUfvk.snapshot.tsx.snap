// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ImportUfvk - test ImportUfvk - snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <Text
        style={
          {
            "color": "rgb(28, 28, 30)",
            "marginTop": 0,
            "opacity": 0.65,
            "padding": 20,
            "textAlign": "center",
          }
        }
      />
      <View
        style={
          {
            "borderColor": "rgb(28, 28, 30)",
            "borderRadius": 10,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "margin": 10,
            "maxHeight": "40%",
            "padding": 10,
          }
        }
      >
        <View
          accessibilityLabel=""
          accessible={true}
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderRadius": 10,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "center",
              "marginRight": 5,
              "width": "auto",
            }
          }
        >
          <TextInput
            editable={true}
            multiline={true}
            onChangeText={[Function]}
            style={
              {
                "backgroundColor": "transparent",
                "color": "rgb(28, 28, 30)",
                "fontSize": 16,
                "fontWeight": "600",
                "marginHorizontal": 5,
                "minHeight": 100,
              }
            }
            value=""
          />
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <
            color="rgb(216, 216, 216)"
            icon={
              {
                "icon": [
                  448,
                  512,
                  [],
                  "f029",
                  "M0 80C0 53.5 21.5 32 48 32h96c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80zM64 96v64h64V96H64zM0 336c0-26.5 21.5-48 48-48h96c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V336zm64 16v64h64V352H64zM304 32h96c26.5 0 48 21.5 48 48v96c0 26.5-21.5 48-48 48H304c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48zm80 64H320v64h64V96zM256 304c0-8.8 7.2-16 16-16h64c8.8 0 16 7.2 16 16s7.2 16 16 16h32c8.8 0 16-7.2 16-16s7.2-16 16-16s16 7.2 16 16v96c0 8.8-7.2 16-16 16H368c-8.8 0-16-7.2-16-16s-7.2-16-16-16s-16 7.2-16 16v64c0 8.8-7.2 16-16 16H272c-8.8 0-16-7.2-16-16V304zM368 480a16 16 0 1 1 0-32 16 16 0 1 1 0 32zm64 0a16 16 0 1 1 0-32 16 16 0 1 1 0 32z",
                ],
                "iconName": "qrcode",
                "prefix": "fas",
              }
            }
            size={48}
          />
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "opacity": 0.65,
            }
          }
        />
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "opacity": 0.65,
              "textAlign": "center",
            }
          }
        >
           (1, --)
        </Text>
        <View
          accessibilityLabel=""
          accessible={true}
          style={
            {
              "borderColor": "rgb(28, 28, 30)",
              "borderRadius": 10,
              "borderWidth": 1,
              "margin": 10,
              "maxHeight": 48,
              "maxWidth": "40%",
              "minHeight": 48,
              "minWidth": "20%",
              "width": "30%",
            }
          }
        >
          <TextInput
            editable={false}
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="#"
            style={
              {
                "backgroundColor": "transparent",
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "marginLeft": 5,
                "minHeight": 48,
                "minWidth": "20%",
              }
            }
            value=""
          />
        </View>
        <Text
          style={
            {
              "color": "rgb(28, 28, 30)",
              "fontSize": 18,
              "fontWeight": "600",
              "margin": 20,
              "marginBottom": 30,
              "opacity": 1,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "marginVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(242, 242, 242)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgb(242, 242, 242)",
          "borderColor": "rgb(0, 122, 255)",
          "borderRadius": 10,
          "borderWidth": 2,
          "justifyContent": "center",
          "marginLeft": 10,
          "maxWidth": "90%",
          "minHeight": 48,
          "minWidth": "30%",
          "opacity": 1,
          "padding": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 0,
            "minHeight": 48,
            "minWidth": 48,
            "padding": 0,
          }
        }
      >
        <Text
          style={
            {
              "color": "rgb(0, 122, 255)",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
